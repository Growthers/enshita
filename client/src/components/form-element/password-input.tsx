import { Icon } from "@iconify/react";
import { forwardRef, useState } from "react";
import styles from "~/styles/components/form-input/info-input.module.scss";
import type { PasswordInputProps } from "./type/model";

// eslint-disable-next-line react/display-name
const PasswordInput = forwardRef<HTMLInputElement, PasswordInputProps>(
  ({ inputStyles, error, ...rest }, ref): JSX.Element => {
    const [isShow, setShow] = useState(true);
    return (
      <>
        <div className={`${styles["enshita-input-textarea"]} ${inputStyles}`}>
          <div className={styles["enshita-input-icon"]}>
            <Icon icon="bxs:lock-alt" width="15" height="15" />
          </div>
          <input
            ref={ref}
            type={isShow ? "password" : "text"}
            className={styles["enshita-input-password"]}
            {...rest}
          />
          <button
            type="button"
            className={styles["enshita-input-icon"]}
            onClick={() => {
              setShow(!isShow);
            }}
          >
            <Icon
              icon={
                isShow
                  ? "ant-design:eye-filled"
                  : "ant-design:eye-invisible-filled"
              }
              width="15"
              height="15"
            />
          </button>
        </div>
        <p className={styles["enshita-input-error"]}>
          {error && <small>{error.message}</small>}
        </p>
      </>
    );
  },
);

export { PasswordInput };
