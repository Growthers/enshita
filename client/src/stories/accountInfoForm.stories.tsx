// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Form from "~/components/form/accountInfoForm";

type T = typeof Form;
type Story = ComponentStoryObj<T>;

export default {
  title: "Account Information Form",
  args: {
    mail: "abc@example.com",
    userName: "hoge",
  },
  component: Form,
} as ComponentMeta<T>;

export const Default: Story = {};
