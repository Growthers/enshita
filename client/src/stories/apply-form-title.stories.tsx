import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { Title } from "~/components/apply-form/title";

type T = typeof Title;
type Story = ComponentStoryObj<T>;

const data = {
  title: "共同開発鯖LT #4",
};

export default { args: { title: data.title }, component: Title } as ComponentMeta<T>;

export const Default: Story = {};
