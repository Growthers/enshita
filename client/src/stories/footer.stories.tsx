// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Footer from "~/components/layout/footer";

type T = typeof Footer;
type Story = ComponentStoryObj<T>;

export default {
  title: "Footer",

  component: Footer,
} as ComponentMeta<T>;

export const Default: Story = {};
