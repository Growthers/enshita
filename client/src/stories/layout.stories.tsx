import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStory } from "@storybook/react";
import Layout from "~/components/layout";

type T = typeof Layout;
type Story = ComponentStory<T>;

export default {
  title: "Layout",

  component: Layout,
} as ComponentMeta<T>;

export const Default: Story = () => (
  <Layout>
    <div>hogehoge</div>
  </Layout>
);
