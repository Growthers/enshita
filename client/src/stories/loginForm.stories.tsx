// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import LoginForm from "~/components/loginForm";

type T = typeof LoginForm;
type Story = ComponentStoryObj<T>;

export default {
  title: "LoginForm",
  component: LoginForm,
} as ComponentMeta<T>;

export const Default: Story = {};
