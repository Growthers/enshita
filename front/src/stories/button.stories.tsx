import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Button from "~/components/button";

type T = typeof Button;
type Story = ComponentStoryObj<T>;

export default {
  title: "Button",
  args: { children: <div>hogehoge</div> },
  component: Button,
} as ComponentMeta<T>;

export const Default: Story = {};
