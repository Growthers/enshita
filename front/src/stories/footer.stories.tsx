import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Footer from "~/components/footer";

type T = typeof Footer;
type Story = ComponentStoryObj<T>;

export default {
  title: "Footer",

  component: Footer,
} as ComponentMeta<T>;

export const Default: Story = {};
