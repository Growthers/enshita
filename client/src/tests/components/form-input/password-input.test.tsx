import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "~/stories/form-input/password-input.stories";

const { Default } = composeStories(stories);

describe("(component) password input module test", () => {
  test("Snap Shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
