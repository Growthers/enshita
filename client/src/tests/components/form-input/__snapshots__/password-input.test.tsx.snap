// Vitest Snapshot v1

exports[`(component) password input module test > Snap Shot 1`] = `
<div>
  <div
    class="_enshita-input-textarea_b32e13 undefined"
  >
    <div
      class="_enshita-input-icon_b32e13"
    >
      <span />
    </div>
    <input
      class="_enshita-input-password_b32e13"
      type="password"
    />
    <button
      class="_enshita-input-icon_b32e13"
      type="button"
    >
      <span />
    </button>
  </div>
  <p
    class="_enshita-input-error_b32e13"
  />
</div>
`;
